### DISCUSSION TITLE
Plugin development [Jun 30]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#6025] [Discord Threads]
> 🙋‍♂️ **melmass**

**Comment by o1lab** [comment]
> 👨‍🦲 **dstala**  
#### Can you elaborate tad more on what you intend to develop?  
#  
> 🙋‍♂️ **melmass**  
#### My goal was mainly extending the frontend, for instance a `type` custom widget.   
#### But I was also curious about the possibilities, if the app store apps relied on the same plugin api for instance? thanks  
#  
> 👨‍🦲 **dstala**  
#### To create something like a dashboard?  
#  
> 🙋‍♂️ **melmass**  
#### More like to add to the noco dashboard directly.   
#### For instance let's say I wanted to have a custom widget for a `file path` type, that in the end with be stored as a string in the DB, but I want a file picker in the UI (this is a dumb example but cover what I meant)  
#  
> 🙋‍♂️ **guyroland**  
#### I'm also interested in this. I wanted to extend the frontend. Any documentation about it <@857998597474484295> ?  
#  
> 👨‍🦲 **dstala**  
#### No specific documentation. Familiarity with TS & VueJS should help with navigating open source code. We will consider adding additional documentation support for such activities in future  
#  
> 🙋‍♂️ **melmass**  
#### Any insight would be good IMO, the template isn't particularly obvious to me, it exports interfaces that I can kind of understand so are those the list of interface a plugin can implement?:   
####    
#### https://github.com/nocodb/nocodb/tree/develop/packages/nc-plugin/src/lib  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1108819597592895571)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
